setup.py

```diff
@@ -14,12 +14,18 @@
     'xmlio': ['uproot'],
     'minuit': ['iminuit'],
 }
-extras_require['test'] = sorted(
+extras_require['backends'] = sorted(
```

Comment:
I guess I understand that you're just adding the `backends` extra, while the `test` extra still has the same effect, but it still feels like this is something that should be a separate PR. Eg, it would be good to be able to need to revert that PR without also reverting these nice docs that were added.  In this proposed additional PR can we also update the [Installation docs for "...with all backends"](https://scikit-hep.org/pyhf/installation.html#with-all-backends) to 

```
pip install pyhf[backends]
```

which is better then what is already there also as it automatically brings in additional backends as we add them to `setup.py`.

Reply:
This is now implemented in the [`refactor/create-backend-extra`](https://github.com/scikit-hep/pyhf/compare/refactor/create-backend-extra) branch. c.f. my edited review comment below.

Reply:
ok. just get that PR merged in first. because those changes need to happen to get the RTD to build.

Reply:
#685